arch/common/isr_tables.c

```diff
@@ -29,20 +29,29 @@ Z_GENERIC_SECTION(.irq_info) struct int_list_header _iheader = {
 /* These are placeholder tables. They will be replaced by the real tables
  * generated by gen_isr_tables.py.
  *
- * z_irq_spurious and _isr_wrapper are used as placeholder values to
- * ensure that they are not optimized out in the first link. The first
- * link must contain the same symbols as the second one for the code
- * generation to work.
+ * z_irq_spurious is used as placeholder values to ensure that they are not
+ * optimized out in the first link. The first link must contain the same
+ * symbols as the second one for the code generation to work.
  */
 
 /* Some arches don't use a vector table, they have a common exception entry
  * point for all interrupts. Don't generate a table in this case.
  */
 #ifdef CONFIG_GEN_IRQ_VECTOR_TABLE
+/* When both the IRQ vector table and the software ISR table are used, populate
+ * the IRQ vector table with the common software ISR by default, such that all
+ * indirect interrupt vectors are handled using the software ISR table.
+ */
+#ifdef CONFIG_GEN_SW_ISR_TABLE
+#define IRQ_VECTOR_TABLE_DEFAULT_ISR	_isr_wrapper
+#else
```

Comment:
Could you also add a comment here?

Reply:
how about an inline comment here at the #else? This seems to be the differentiation brought in this patch

Reply:
I am not sure if I understand what you mean by "inline comment" here.

Reply:
Well I'd like a comment as the one in lines 41-44, for the #else case, i.e. when only the vector tables are used.

Reply:
I thought it was quite self-explanatory -- that unconnected IRQs should have `z_irq_spurious`, as you would in `_sw_isr_table`.

I will add a comment explaining this anyway, since this might not be as obvious to the person reading this for the first time.

Reply:
yes, please

Reply:
```
/* When both the IRQ vector table and the software ISR table are used, populate
 * the IRQ vector table with the common software ISR by default, such that all
 * indirect interrupt vectors are handled using the software ISR table;
 * otherwise, populate the IRQ vector table with z_irq_spurious so that all
 * un-connected IRQ vectors end up in the spurious IRQ handler.
 */
```